core/domain/state_domain.py

```diff
@@ -1388,20 +1397,59 @@ def update_interaction_id(self, interaction_id):
         Args:
             interaction_id: str. The new interaction id to set.
         """
+        if interaction_id is None:
+            unwanted_content_ids = self._get_customization_args_content_ids()
```

Comment:
You should probably define _get_customization_args_content_ids() on the Interaction object?

Reply:
I think interaction Object just have the specs and static properties into it, I didn't find any dynamic things there. (customization_args spec is static but customization_args are dynamic.)

Reply:
No, not the interaction specs. I meant the Interaction class in state_domain.